Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml

```diff
@@ -22787,9 +22787,11 @@
     </CActorModel>
     <CActorModel id="AP_SquadSightAttachment" parent="ModelAnimationStyleContinuous">
         <Inherits index="Visibility" value="1"/>
+        <On Terms="Effect.AP_SquadSightPersistent.Start; At Caster" Send="Create"/>
+        <On Terms="Effect.AP_SquadSightPersistent.Stop; ValidateUnit AP_NotHaveHavocEnduringSight; At Caster" Send="AnimBracketStop BSD"/>
         <On Terms="Behavior.AP_SquadSightPersistentBehavior.On" Send="Create"/>
         <On Terms="Behavior.AP_SquadSightPersistentBehavior.Off" Send="AnimBracketStop BSD"/>
-        <On Terms="ActorCreation; ValidateUnit AP_HaveAlarakHavocAbilityRange" Send="SetScale 1.75"/>
+        <On Terms="Upgrade.AP_AlarakHavocAbilityRange.Add" Send="SetScale 1.75"/>

```

Comment:
The actor shows the short range if the squadsight range upgrade is researched, regardless of war council upgrade:
![Snímek obrazovky pořízený 2024-07-25 19-49-36](https://github.com/user-attachments/assets/a9d9813a-d6df-479f-a02a-6eda72c54e96)

Reply:
Also the red range actor shows up even if the ability isn't activated

Reply:
Hm I remember adding that line and then removing it later because it seemed to be unnecessary and also made the scale huge. Will tinker a bit.

Reply:
Seems when I did my last test, I deleted the wrong line. Old line should be restored, and used the template of the AP_Monitor actor sending messages to the other actors on upgrade.add to get things mostly working.